.github/workflows/cibuildwheel.yml

```diff
@@ -136,31 +136,31 @@ jobs:
           path: dist/
 
       - name: Build wheels for CPython 3.11
-        uses: pypa/cibuildwheel@v2.13.0
+        uses: pypa/cibuildwheel@51f5c7fe68ff24694d5a6ac0eb3ad476ddd062a8 # v2.13.0
```

Comment:
@QuLogic I'm not actually sure if this is what you were suggesting in https://github.com/matplotlib/matplotlib/pull/26023#discussion_r1212575095 or not. Having the commit SHA for the `pypa/cibuildwheel` action adds security around the wheel build, but as your upload is not to a package index but to GitHub's artifact store (so probably less of a security risk here as you have inspection ability post upload) with

https://github.com/matplotlib/matplotlib/blob/bfaa6eb677b9c56cafb6a99d6897c9d0cd9d4210/.github/workflows/cibuildwheel.yml#L171-L175

I'm not sure how `matplotlib` finally publishes wheels to PyPI and if that workflow should have additional hardening.

cc @ksunden

Reply:
we download them and upload manually.  In the past people have gotten very bent out of shape if the sdist goes up before the wheels so a human ensures they are sequenced right.

Reply:
> In the past people have gotten very bent out of shape if the sdist goes up before the wheels so a human ensures they are sequenced right.

:grimacing: Sorry to hear that, but thanks for the explanation.